Update report-a-bug (#568)
ubuntu/ubuntu-project-docs: docs/contributors/qa-and-testing/report-a-bug.md

Reviewer: This seems like a prerequisite to me, can we move it closer to the top of the page? (or, above the previous section anyway)

Author: The current order makes sense to me.  I'm not sure re-ordering makes senses
1. Intro.
1. Bugs are on launchpad. what's launchpad is
1. create a launchpad account

(how-to-report-bugs)=
## How to report bugs

> Ubuntu uses {ref}`Launchpad <about-launchpad>` to keep track of bugs and their fixes. This page will guide you through the steps required to file a good and detailed report.

Reviewer: ```suggestion
Ubuntu uses {ref}`Launchpad <about-launchpad>` to keep track of bugs and their fixes.
```

Reviewer: Looking at the rendered page, this line stands out as a duplication of the last line of the paragraph above it. It would be good to remove it while we're touching the page anyway.

> When possible, this will involve running a command `ubuntu-bug` alias `apport-bug` alias `apport-collect` Note that this only applies to traditional .deb packages.

Reviewer: ```suggestion
This involves running a command `ubuntu-bug` (aliases `apport-bug` and `apport-collect`). Note, this only applies to traditional .deb packages.
```

(determine-if-the-bug-is-really-a-bug)=
## Determine if the bug is really a bug

You should **not** file a bug if you are:

* **Requesting new software:** You should follow the guidelines at {ref}`New Packages<new-packages>`.

* **Requesting support:** See the list of {ref}`community-support` resources.

* **Discussing features, existing policy, proposing features, or ideas:** This belongs to the [ubuntu-devel-discuss](https://lists.ubuntu.com/mailman/listinfo/ubuntu-devel-discuss) mailing list.

> * **Using a package not provided by the default, supported Ubuntu repositories:** Unsupported software can cause a variety of issues, even to the operating system. For this reason, you should try reproducing the issue on systems which only have supported software installed to increase the chances your bug will be addressed. Typically, what isn't supported is software from a PPA (Personal Package Archive), 3rd party packages, self-compiled software, etc. For more on supported Ubuntu repositories, please see {ref}`package-archive <package-archive>`. If you are using unsupported software, it is best to contact the maintainers directly. Instructions are generally available on the program maintainer's website.

Reviewer: ```suggestion
* **Using a package not provided by the default, supported Ubuntu repositories:** Unsupported software can cause a variety of issues, even to the operating system. For this reason, you should try reproducing the issue on systems which only have supported software installed to increase the chances your bug will be addressed. Typically, what isn't supported is software from a Personal Package Archive (PPA), 3rd party packages, self-compiled software, etc. For more on supported Ubuntu repositories, please see {ref}`package-archive <package-archive>`. If you are using unsupported software, it is best to contact the maintainers directly. Instructions are generally available on the program maintainer's website.
```

Author: I beg to differ on that one. The spelled-out version is rarely used. Users will see PPA almost everywhere.

Reviewer: It's become standard to have acronyms be defined in this format on first use on every page, mainly for accessibility best practice. The spelled-out version may not be often used by those who are already maintainers, but for those who are *not* maintainers yet, or are not developers at all (the primary target audiences for this page), we can't safely assume they already know what PPA stands for.

Author: my point is that I don't think most regular users know the spelled-out version, as a lot of tooling, manpages, and Launchpad usually refer to them as PPA. See [search trends](https://trends.google.com/explore?q=Personal%2520Package%2520Archive%2CPPA&date=today%205-y&geo=Worldwide).

Anyhow, it's not worth debating this. I like consistency :smiley: . Fixing…

(perform-a-survey-of-your-problem)=
## Perform a survey of your problem

First, check the [release notes](https://documentation.ubuntu.com/release-notes/) of your supported version of Ubuntu for any known issues.

> Second, check [Launchpad](https://bugs.launchpad.net/) for any duplicates, and make note of this.

Reviewer: ```suggestion
Second, check [Launchpad](https://bugs.launchpad.net/) for any duplicate bugs or issues, and make note of this.
```

(reporting-a-crash)=
## Reporting a crash

> If an application crashes, what typically happens is [Apport](../debugging/apport) will display a window noting it is collecting information about the crash:

Reviewer: ```suggestion
If an application crashes, what typically happens is {ref}`debugging-apport` will display a window noting it is collecting information about the crash:
```

(collecting-information-from-a-specific-package)=
### Collecting information from a specific package

Press Alt+F2 to open the “Run Command” screen:

![Filing a bug with the “Run Command” screen](gnome-ubuntu-bug-pkgname.png)

Then, type `ubuntu-bug <package name>` and press Enter. If you’re not sure which package has the problem, refer to the instructions for [finding the right package](https://wiki.ubuntu.com/Bugs/FindRightPackage).

(collecting-information-from-a-currently-running-program)=
### Collecting information from a currently running program

To file a bug against a program that is currently running, open the System Monitor application and find the ID of the process.

![System Monitor Processes tab](system-monitor-id-new.png)

Then type `ubuntu-bug ` followed by the process ID into the “Run Command” screen.

![Filing a bug with the “Run Command” screen and a process ID](gnome-ubuntu-bug-pid.png)

(filing-a-general-bug-against-no-particular-package)=
## Filing a general bug against no particular package

> First, please review potential package candidates [here](../bug-triage/assign-a-bug-to-a-package). Only after reviewing this, if are still not sure which package is affected by the bug, type `ubuntu-bug` in the “Run Command” screen and press Enter. This will guide you through a series of questions to gather more information about the bug and help you assign it to the appropriate package.

Reviewer: ```suggestion
First, please {ref}`review potential package candidates <how-to-assign-a-bug-to-a-package>`. Only after reviewing this, if you are still not sure which package is affected by the bug, type `ubuntu-bug` in the “Run Command” screen and press Enter. This will guide you through a series of questions to gather more information about the bug and help you assign it to the appropriate package.
```

Fill in the description field with as much information as you can, it is better to have too much information in the description than not enough.

At then bottom of the page, there are some extra options you can use to make your bug report more complete:

* **This bug is a security vulnerability:** Please check this **_only_** if your bug report describes a behavior that could be exploited to compromise your security or safety, as well as cause issues such as identity theft or "hijacking".

* **Tags:** You can {ref}`add here tags <bug-tags>` that pertain to your bug report. The predefined values should be left alone.

> * **Include an attachment:** Using this option, you can add supporting attachments to explain or help others reproduce the bug. This might include a screenshot, a video capture of the problem or a sample document that triggers the fault. If necessary, additional attachments can be added after the bug is reported via **Add a comment/attachment** at the bottom of the page. Please check [Debug an apport crash](../debugging/debug-an-apport-crash) for any further information to provide. It is vital for developers to get this information, as it contains the minimum requirement information necessary for a developer to begin working on your bug.

Reviewer: ```suggestion
* **Include an attachment:** Using this option, you can add supporting attachments to explain or help others reproduce the bug. This might include a screenshot, a video capture of the problem or a sample document that triggers the fault. If necessary, additional attachments can be added after the bug is reported via **Add a comment/attachment** at the bottom of the page. Please check {ref}`how-to-debug-an-apport-crash` for any further information to provide. It is vital for developers to get this information, as it contains the minimum information necessary for a developer to begin working on your bug.
```

ubuntu-bug FILENAME.apport
```

Please do not attach the `.apport` or `.crash` file to the report, as this is not the same as performing the above-mentioned steps.

(filing-bugs-manually-at-launchpad-net)=
### Filing bugs manually at Launchpad.net

Before you proceed, you should think about the nature of the problem you're facing. If Ubuntu or its software seems to simply be generally "misbehaving", it might not be a software bug, but it is still a problem we want to help fix. We have an entire community of people who can help you in real time on our  {ref}`Matrix channels and Discourse forums <community-support>`. On the other hand, if you are absolutely sure that you've encountered a legitimate error in the software's behavior (especially one that can be reproduced regularly), and you're sure you don't need any help, then continue with the bug reporting process.

> If for some reason you cannot file a bug using the *Apport* tool you can file one via [Launchpad's own bug report form](https://bugs.launchpad.net/ubuntu/+filebug/+login?no-redirect). When doing so it is best if you have determined which package it should be filed against. Read '[finding the right package](../bug-triage/assign-a-bug-to-a-package)' for guidance or use [Launchpad's package search feature](https://launchpad.net/ubuntu/). We don't recommend this method for most bug reports because they will likely be missing crucial information, use `ubuntu-bug` if you can!

Reviewer: ```suggestion
If for some reason you cannot file a bug using the *Apport* tool you can file one via [Launchpad's own bug report form](https://bugs.launchpad.net/ubuntu/+filebug/+login?no-redirect). When doing so it is best if you have determined which package it should be filed against. Read {ref}`how-to-assign-a-bug-to-a-package` for guidance or use [Launchpad's package search feature](https://launchpad.net/ubuntu/). We don't recommend this method for most bug reports because they will likely be missing crucial information, use `ubuntu-bug` if you can!
```

### All bug reports

* **Please do not file bug reports about End-of-Life operating systems.**

>    This includes release of Ubuntu, and alternative operating systems. Expecting Ubuntu to provide interoperability with an insecure, end-of-life operating system is simply irresponsible, and inconsiderate of the finite resources of the Ubuntu Community. See information regarding [supported Ubuntu releases](https://documentation.ubuntu.com/project/release-team/list-of-releases/). Please see the website of the vendor of the operating system for EOL and support information.

Reviewer: ```suggestion
   This includes release of Ubuntu, and alternative operating systems. Expecting Ubuntu to provide interoperability with an insecure, end-of-life operating system is simply irresponsible, and inconsiderate of the finite resources of the Ubuntu Community. See information regarding {ref}`list-of-releases`. Please see the website of the vendor of the operating system for EOL and support information.
```

* **Please do not speculate on what you think is or isn't a duplicate report**
   The exception to this is you are a developer, know specifically where in the code the problem is, and would be submitting a patch to fix the issue. However, noting things like, "I checked Google and found bug report number...", "Why should I file a new report when this is a duplicate?" is largely unhelpful as it tends not to be a duplicate, and already has been or easily done by triagers and developers. Instead, if you are the original reporter, ensuring the report has all the requested testing information performed would be the fastest way to ensure your bug is resolved as soon as possible. If you are not the original reporter, it's best to file a new report, so that necessary debugging attachments are reviewed. It is a common misconception that filing what one initially believes to be a potential duplicate report is not helpful. Filing a new report is quite helpful, and is preferred to ease triaging and development.

> * **Please do not quote Wikipedia and other non-primary resource information as fact on Launchpad.**.

Reviewer: ```suggestion
* **Please do not quote Wikipedia and other non-primary resource information as fact on Launchpad.**
```

* **Please do not complain about how long it takes to fix a bug.**
   This goes along with saying things like severity of your bug is high so it should be fixed immediately, “I cannot believe it’s not fixed…”, XYZ person(s) do not care about fixing bugs, etc. Especially, if you have not followed the directions mentioned in this article, let alone contributed code upstream. This type of behavior is nonconstructive, irritating to others who read your e-mail, and spammy. We all want to see every bug fixed as soon as possible! Naturally, bugs being fixed is limited to reproducibility and clarity of the bug report, the actual impact the bug has on the community, and available developer resources.

* **Please keep the bug report as objective as possible.**
   It is desired for you to provide a fact based, technical impact statement on you, your environment, and the potential or actual impact on the community at large.

> * **Please provide all relevant information from [Debug an apport crash](../debugging/debug-an-apport-crash) when you first report your bug**.

Reviewer: ```suggestion
* **Please provide all relevant information from {ref}`how-to-debug-an-apport-crash` when you first report your bug**.
```
